test/sequential/test-fs-readfile.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+const common = require('../common');
+
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+
+common.refreshTmpDir();
```

Comment:
I'm a bit lost on how this test verifies that the files are read in chunk.

Reply:
This is a test for the correctness of `fs.readFile`, as I attempted to indicate by its name and its placement in `test/sequential`. There was no test for this API so I added one in order to ensure that my changes did not introduce a regression.

Reply:
Can you please add a short explanation before the test?

Why is it in sequential?

Reply:
> Can you please add a short explanation before the test?

Sure. I looked through a few tests and didn't see any explanations, so can you point me to a good example of this in `test/`?

> Why is it in sequential?

Seemed reasonable at the time. Where should it be instead? `parallel/`?

Reply:
> Sure. I looked through a few tests and didn't see any explanations, so can you point me to a good example of this in test/?

It's true that a lot of tests don't have such a comment, but we've been trying to get better about adding them lately The short example test file shown in https://github.com/nodejs/node/blob/master/doc/guides/writing-tests.md is a good model, I think.

> Where should it be instead? parallel/?

`parallel` is the default place we put things. If they can't be safely run while other tests are run, then we use `sequential`. (For example, if a test uses a hard-coded network port number like `common.PORT`, it should be in `sequential` because otherwise it may have a port collision with another test. Many tests in `parallel` use port `0` which is "give me an arbitrary open port", and if one of those tests grabs `common.PORT` first, a test using `common.PORT` may fail with `EADDRINUSE` or something similar.)

Reply:
@Trott Great explanation, thank you. I'll fix this soon.

Reply:
Fixed in 653052e.

I introduced more paranoid fill to catch bugs specific to more complex `fs.readFile` implementations like the partitioning proposed in this PR.